setup.py

```diff
@@ -52,7 +48,9 @@ def get_dependencies(requirements_path=None):
     'scipy',
     'scikit-image',
     'scikit-learn',
-    'tensorflow~=1.15.0',
+    'intel-tensorflow~=2.7.0; sys_platform != "darwin"',
+    'tensorflow~=2.7.0; sys_platform == "darwin"',
```

Comment:
@joshuacwnewton do you happen to know if there's a MacOS TF2 version with MKL support? Right now the vanilla CPU version can't run ConvXD layers correctly :(

I got around this issue for Windows and Linux by using `intel-tensorflow`. Unfortunately I don't have a Mac machine that I can legally use for this purpose.

Reply:
Current `tensorflow-metal` package requires Python 3.{8,9} so using it is out of the question :( Looking for alternatives for `tensorflow` accelerations

Reply:
One small note:: The only thing keeping us from upgrading to Python 3.8/3.9 _is_ TensorFlow (2.2 and 2.5 respectively). (https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3367)

So, upgrading both at the same time wouldn't be out of the question!

Reply:
One other comment I want to make: In SCT, we're merely using TF/Keras to perform inference using pre-trained models (as opposed to training models from scratch). So, we actually don't have much of a need for GPU-acceleration? In fact, it may even be preferable to stick to CPU-only libraries, since that would typically reduce the installation size of SCT.

So, if you have a working CPU-based upgrade to TF, then I would say that's more than good enough. :)

Reply:
Here's my attempt at summarizing the exploration you've done (just to make sure I'm understanding OK):

---

Upgrading stock `tensorflow` to 2.X causes a error to be thrown re: `NHWC` vs. `NCHW` data formatting. Specifically:

```
CPU implementation of Conv3D currently only supports the NHWC tensor format.
```

(Ah! I'm just realizing I came across the exact same error in my previous investigations, too: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3367#issuecomment-907357925)

This is the root issue, and there are [three possible workarounds](https://github.com/onnx/onnx-tensorflow/issues/535#issuecomment-568743917) for this issue (according to [the linked comment](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3653#discussion_r787271777), anyway):

  1. Using an MKL-based version of Tensorflow:
      * [`intel-tensorflow`](https://pypi.org/project/intel-tensorflow/) is a pip package that provides MKL support, but only for Linux/Windows, and not macOS. 
      * [`tensorflow-mkl`](https://anaconda.org/anaconda/tensorflow-mkl) is a conda package that provides MKL support for macOS, but only for TF 2.0.0. (Dead-end for macOS support.)
  2. Using a GPU-based version of Tensorflow:
       * [`tensorflow-metal`](https://pypi.org/project/tensorflow-metal/) is a pip package that provides GPU support for macOS, but only for Python 3.8/3.9. (And it would also [involve PluggableDevices](https://pypi.org/project/tensorflow-metal/), which may require more complicated prerequisites on the user side of things.)
   3. Trying to... modify the data format on the model side of things?
       * As far as I can tell, we haven't explored this option yet, right?

Reply:
There's a potential "workaround 4." in [this comment](https://github.com/onnx/onnx-tensorflow/issues/535#issuecomment-683366977):

> setting `os.environ["CUDA_VISIBLE_DEVICES"]="-1"` (as mentioned in this reply [#614 (comment)](https://github.com/onnx/onnx-tensorflow/issues/614#issuecomment-625251872)) fixes the issue for me.

That linked reply (from onnx-tensorflow issue 614) seems pretty illuminating to me! I didn't really understand _why_ the NWHC/NCHW formatting issues were a problem in the first place, but that explanation maybe clarifies things a little? :eyes:

Reply:
That's a good summary of the situation. For "workaround 4", I don't think it works because `tensorflow-cpu` package fails as well.